modules/openapi-generator/src/main/resources/python/api_response.mustache

```diff
@@ -12,17 +18,56 @@ class ApiResponse:
     status_code: Optional[StrictInt] = Field(None, description="HTTP status code")
     headers: Optional[Dict[StrictStr, StrictStr]] = Field(None, description="HTTP headers")
     data: Optional[Any] = Field(None, description="Deserialized data given the data type")
-    raw_data: Union[str, bytes] = Field(description="Raw data (HTTP response body)")
+    _raw_data: Union[None, str, bytes] = Field(description="Raw data (HTTP response body)")
+    {{#asyncio}}
+    aiohttp_response: ClientResponse = Field(description="Raw aiohttp response for response streaming")
+    {{/asyncio}}
+    {{^asyncio}}
+    urllib3_response: HTTPResponse = Field(description="Raw urllib3 response for response streaming")
+    {{/asyncio}}
 
     def __init__(
         self,
         status_code=None,
         headers=None,
         data=None,
         *,
-        raw_data,
+        _raw_data,
+        {{#asyncio}}
+        aiohttp_response,
+        {{/asyncio}}
+        {{^asyncio}}
+        urllib3_response,
+        {{/asyncio}}
     ) -> None:
         self.status_code = status_code
         self.headers = headers
         self.data = data
-        self.raw_data = raw_data
+        self._raw_data = _raw_data
+        {{#asyncio}}
+        self.aiohttp_response = aiohttp_response
+        {{/asyncio}}
+        {{^asyncio}}
+        self.urllib3_response = urllib3_response
+        {{/asyncio}}
+
+    @property
+    def raw_data(self) -> Union[str, bytes]:
+        """Get raw response body.
+
+        If `preload_content=False`, this loads the entire response body into
+        memory. If `preload_content=True`, the response body is already loaded
+        earlier.
+        To stream responses, read from `self.{{#asyncio}}aiohttp{{/asyncio}}{{^asyncio}}urllib3{{/asyncio}}_response`.
+
+        :return: `str` if `preload_content=True`, `bytes` if not.
+        """
+        if self._raw_data is None:
+            {{#asyncio}}
+            loop = asyncio.get_event_loop()
+            return loop.run_until_complete(self.aiohttp_response.read())
+            {{/asyncio}}
+            {{^asyncio}}
+            return self.urllib3_response.data
+            {{/asyncio}}
```

Comment:
I think `ApiResponse.raw_data` should be `None` if `_preload_content=True`, but that would be a (avoidable) breaking change. So I added this ugly code for backwards-compatibility.

We could eventually offer an extra `ApiResponse.read()` method, which can be properly async.

Should I raise a `DeprecationWarning` in this code path? Then we should provide an `ApiResponse.read()` method or tell users to read from the backend objects.

Reply:
The `read()` method would be needed for the asyncio client, it's not possible to have an `async` property and "hide" the loading logic into the property code itself.

So yeah, I would say the deprecation makes sense to me.

Reply:
I will set `raw_data` to `None` if `_preload_content=False` in the aiohttp client (see https://github.com/OpenAPITools/openapi-generator/pull/16789#discussion_r1357853586).

For the urllib3 client, I will keep returning `self.urllib3_response.data` if `_preload_content=False`, but raise a deprecation warning from now on. 

And I'll add the `read()` method.

On the next major release, we can make `raw_data` always be `None` if `_preload_content=False`, so users have to consciously load data with the new `read()` method.